checker/src/visitors.rs

```diff
@@ -2084,23 +2084,24 @@ impl<'analysis, 'compilation, 'tcx, E> MirVisitor<'analysis, 'compilation, 'tcx,
                 }
             }
             KnownFunctionNames::MiraiPostcondition => {
-                if self.post_condition.is_some() {
-                    let span = self.current_span;
-                    let warning = self
-                        .session
-                        .struct_span_warn(span, "only one post condition is supported");
-                    self.emit_diagnostic(warning);
-                }
+                let extend_post_condition = |current: &Option<Rc<AbstractValue>>, cond| {
+                    if let Some(current_cond) = current {
+                        Some(current_cond.and(cond))
```

Comment:
This and is not safe as is. In the test case below it will result in the a post condition that states `result < 5 && result > 10.` The desired post condition is instead `if c { result < 5 } else { result > 10 }`. Of course, doing the latter simplistically is also problematic because if might end up with things like `if c { result < 5 } else if c { result > 2 } else { result > 10 }.` Obviously we would want that to be `if c { result < 5 && result > 2 } else { result > 10 }`.